lfs/scanner.go

```diff
@@ -192,7 +195,8 @@ func (m *indexFileMap) Set(sha string, index *indexFile) {
 // ScanIndex returns a slice of WrappedPointer objects for all
 // Git LFS pointers it finds in the index.
 // Reports unique oids once only, not multiple times if >1 file uses the same content
-func ScanIndex() ([]*WrappedPointer, error) {
+// isFirstCommit must be true if this repo doesn't have any commits yet
+func ScanIndex(isFirstCommit bool) ([]*WrappedPointer, error) {
```

Comment:
I wanted callers to ScanIndex not to have to know exactly what they needed to do to support pre-initial commit but you're right that when reading the code afterwards just seeing `false` is less useful than seeing `InitialCommit` so I'll do that. 

`HeadRef` would be good in theory but because that would introduce an extra `git` call just to derive `HEAD` in 99% of cases, and when most of the time it actually makes sense to abort before the first commit - `status` is one of the few where it makes sense to do something useful - I won't do this for now. If you can think of some other cases where it's useful to get the null tree automatically we can add it?